python/modem-firmware-1.3+/device_credentials_installer.py

```diff
@@ -438,44 +465,58 @@ def main():
-    if args.gateway:
-        is_gateway = True
-    print(local_style('Opening port {} as {}...'.format(port,
-                                                 'gateway' if is_gateway
-                                                 else 'generic device')))
-
-    # try to open the serial port
-    try:
-        ser = serial.Serial(port, 115200, xonxoff= args.xonxoff, rtscts=(not args.rtscts_off),
-                            dsrdtr=args.dsrdtr, timeout=serial_timeout)
-        ser.reset_input_buffer()
-        ser.reset_output_buffer()
-    except serial.serialutil.SerialException:
-        sys.stderr.write(error_style('Port could not be opened; not a device, or open already\n'))
-        sys.exit(2)
-
-    # for gateways, get to the AT command prompt first
-    if is_gateway:
-        print(local_style('Getting to prompt...'))
-        handle_login()
+    if args.rtt:
+        cmd_term_key = 'CRLF'
 
-        print(local_style('Disabling logs...'))
-        write_line('log disable')
-        wait_for_prompt()
+        rtt = rtt_interface.connect_rtt(args.jlink_sn, args.mosh_rtt_hex)
+        if not rtt:
+            sys.stderr.write(error_style('Failed connect to device via RTT'))
+            sys.exit(2)
 
-        print(local_style('Getting to AT mode...'))
-        write_line('at enable')
-        wait_for_prompt(b'to exit AT mode')
+        if not rtt_interface.enable_at_cmds_mosh_rtt(rtt):
+            sys.stderr.write(error_style('Failed to enable AT commands via RTT'))
+            sys.exit(2)
+        ser = None
+    else:
+        # get a serial port to use
+        port = ask_for_port(args.port, args.all)
+        if port == None:
+            sys.exit(1)
+
+        # let user know which we are using and as what kind of device
+        if args.gateway:
+            is_gateway = True
+        print(local_style('Opening port {} as {}...'.format(port,
+                                                    'gateway' if is_gateway
+                                                    else 'generic device')))
+
+        # try to open the serial port
+        try:
+            ser = serial.Serial(port, 115200, xonxoff= args.xonxoff, rtscts=(not args.rtscts_off),
+                                dsrdtr=args.dsrdtr, timeout=serial_timeout)
+            ser.reset_input_buffer()
+            ser.reset_output_buffer()
+        except serial.serialutil.SerialException:
+            sys.stderr.write(error_style('Port could not be opened; not a device, or open already\n'))
+            sys.exit(2)
+
+        # for gateways, get to the AT command prompt first
+        if is_gateway:
+            print(local_style('Getting to prompt...'))
+            handle_login()
+
+            print(local_style('Disabling logs...'))
+            write_line('log disable')
+            wait_for_prompt()
+
+            print(local_style('Getting to AT mode...'))
+            write_line('at enable')
+            wait_for_prompt(b'to exit AT mode')
```

Comment:
Consider moving this behavior to a `serial_interface.py` with symmetric purpose to `rtt_interface.py`